pandas/core/indexes/datetimes.py

```diff
@@ -383,6 +383,10 @@ def union_many(self, others):
                 this = this._fast_union(other)
             else:
                 this = Index.union(this, other)
+
+        res_name = get_unanimous_names(self, *others)[0]
+        if this.name != res_name:
+            return this._shallow_copy(name=res_name)
```

Comment:
rename instead of shallow_copy?

Reply:
yup, indeed. i actually went through and made the same change in `base.py` and `api.py`. seems like it works fine all around.